industry_benchmarks/utils/fix_networks.py

```diff
@@ -0,0 +1,649 @@
+    missing_ligands = list(
+        input_ligand_network.nodes.difference(res_ligand_network.nodes)
+    )
+
+    # turn the missing ligands into spare ligand networks
+    for missing_ligand in missing_ligands:
+        ligand_sub_networks.append(LigandNetwork(nodes=[missing_ligand], edges=[]))
+
+    # Sort the sub-networks by their size in order of biggest to smallest
+    ligand_sub_networks = list(
+        sorted(ligand_sub_networks, key=lambda sn: len(sn.nodes), reverse=True)
+    )
+
+    # check status
+    print("LOG: Reporting input file contents: ")
+    print(f"\tPlanned input  no. ligands: {len(input_ligand_network.nodes)}\n")
+    print(f"\tPlanned input  no. connections: {len(input_ligand_network.edges)}\n")
+    print(f"\tSimulation results no. ligands: {len(res_ligand_network.nodes)}\n")
+    print(f"\tSimulation results no. connections: {len(res_ligand_network.edges)}\n")
+    print(f"\tMissing ligands in simulation results: {len(missing_ligands)}\n")
+    #ToDo: If there is no disconnected network this would return 1 which is confusing
+    print(f"\tDisconnected networks which need patching: {len(ligand_sub_networks)}\n")
+    print(
+        f"\tLigands in each disconnected network: {[len(sn.nodes) for sn in ligand_sub_networks]}"
+    )
+    print("\n")
+
+    if len(ligand_sub_networks) == 1:
+        # Return gracefully, don't fail!
+        message = ("Did not find disconnected components in alchemical "
+                   "network, nothing to do here!")
+        print(message)
+        return None
+
+    print()
+
+    # Tape the networks together
+    print("LOG: finding additional connections to merge the broken networks:")
+    network_connections = get_new_network_connections(
+        ligand_sub_networks=ligand_sub_networks,
+        input_ligand_network=input_ligand_network,
+    )
+
+    print(f"\tLOG: Generated new ligand transform edges: {len(network_connections.edges)}\n")
+
+    # write out
+    print("LOG: writing out network of additional edges to fix the network:")
+    output_alchemical_network_folder.mkdir(exist_ok=False, parents=True)
+    taped_alchemical_network = get_fixed_alchemical_network(
+        network_connections, input_alchem_network
+    )
+    alchemical_network_json_fp = (
+        output_alchemical_network_folder / "alchemical_network.json"
+    )
+    json.dump(
+        taped_alchemical_network.to_dict(),
+        alchemical_network_json_fp.open(mode="w"),
+        cls=JSON_HANDLER.encoder,
+    )
+
+    # Write out each transformation
+    # Create a subdirectory for the transformations
+    transforms_dir = pathlib.Path(output_alchemical_network_folder / "transformations")
+    transforms_dir.mkdir(exist_ok=True, parents=True)
+
+    for transform in taped_alchemical_network.edges:
+        transform.dump(transforms_dir / f"{transform.name}.json")
+
+    full_ligand_network = get_full_ligand_network(input_ligand_network, res_ligand_network, network_connections)
```

Comment:
Change this to be only the additional edges.

Reply:
Addressed in commit 31c2ac1